### DISCUSSION TITLE
Migrating Sophia Script settings to a new version

### DISCUSSION THREAD

**Discussion by juanriccio** [farag2/Sophia-Script-for-Windows#245] [Q&A] [answered]
Hello,
I wonder: how can I migrate my carefully crafted settings (in the Sophia.ps script) from one (old) version of Sophia Script to a newer one? I had to do this just once, and I used the output of
         `diff Sophia-crafted.ps1 Sophia.ps1`
where `Sophia-crafted.ps1` is my edited version (from the old version) and `Sophia.ps1` is the originally distributed version (also from the old version).
I then replicated my choices, as shown by `diff`, on the `Sophia.ps1` distributed with the new version. I did it by hand, with two editor windows open.

I'm sure there must be an easier way...

**Comment by BenchTweakGaming** [comment, marked as answer]
Hello Juan,
You can try to 'Import Preset' using the wrapper to import your 'crafted' script. It should fill up the selections (checkboxes, combobox items). Then you can go to console output tab and see a new created script from your choices and save that using 'save as' into a new script.

Remember, sometimes some things get renamed and changed (function names and tags) in updated versions therefore not all things will be imported properly (not one to one). 

Farag2/Dmitry is always updating everything so please keep up to date and go through the script with a fine comb.

**Reply by juanriccio** [reply]
Thanks for the replies! @BenchTweakGaming. I knew things get renamed and changed, hence my question in the first place. Your suggestion to try importing the preset first is useful, as is @farag2's hint about a more powerful comparison tool.

**Comment by farag2** [comment]
The only thing I may advice is to use some kind of the Beyond Compare app, as I do.
